hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/metrics/TestNameserviceRPCMetrics.java

```diff
@@ -134,5 +136,26 @@ public void testProxyOp() throws IOException {
         getMetrics(NAMESERVICE_RPC_METRICS_PREFIX + "ns1"));
   }
 
+  @Test
+  public void testProxyOpCompleteConcurrent() throws IOException {
+
+    long ns0ProxyOpBefore = getLongCounter("ProxyOp",
+        getMetrics(NAMESERVICE_RPC_METRICS_PREFIX + "ns0"));
+    long ns1ProxyOpBefore = getLongCounter("ProxyOp",
+        getMetrics(NAMESERVICE_RPC_METRICS_PREFIX + "ns1"));
+    assertCounter("ProxyOp", 0L,
+        getMetrics(NAMESERVICE_RPC_METRICS_PREFIX + "concurrent"));
+    // RPC which uses invokeConcurrent.
+    router.getRpcServer().setBalancerBandwidth(1048576L);
+
+    assertCounter("ProxyOp", ns0ProxyOpBefore + 1,
+        getMetrics(NAMESERVICE_RPC_METRICS_PREFIX + "ns0"));
+    assertCounter("ProxyOp", ns1ProxyOpBefore + 1,
+        getMetrics(NAMESERVICE_RPC_METRICS_PREFIX + "ns1"));
+    assertCounter("ProxyOp", 1L,
+        getMetrics(NAMESERVICE_RPC_METRICS_PREFIX + CONCURRENT));
+
```

Comment:
Can you also check that the proxy time for the "concurrent" namespace is being created.

Reply:
done.

Reply:
Sorry I wasn't clear here. I meant to check that ProxyAvgTime has changed. Thinking more about this, that's not really necessary since both are updated at the same place here : https://github.com/apache/hadoop/blob/trunk/hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/metrics/NameserviceRPCMetrics.java#L125

So the current PR looks good to me.